### ISSUE TITLE
Fetching origin hangs when macbook returns from sleep

### ISSUE THREAD

**Issue by arnassavickas** [desktop/desktop#18482]
### The problem

Sometimes when I wake my Macbook Pro M1 2019 from sleep, Github Desktop app has an infinitely spinning `Fetching origin` button
<img width="226" alt="image" src="https://github.com/desktop/desktop/assets/33692872/a90453d6-4ae3-4aae-826d-54391068e282">

The only solution is to close the app and reopen it.

This has been happening for a year already with all recent app versions.

### Release version

Version 3.3.13 (arm64)

### Operating system

MacOS Sonoma 14.4.1 (23E224)

### Log files

[2024-04-18.desktop.production.log](https://github.com/desktop/desktop/files/15033601/2024-04-18.desktop.production.log)
[2024-04-19.desktop.production.log](https://github.com/desktop/desktop/files/15033602/2024-04-19.desktop.production.log)

**Comment by steveward** [comment]
Thanks for the report @arnassavickas. I can see the background fetch hanging in the log:

`2024-04-19T05:41:41.386Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 15 repositories took 43.1s of which 69100.2s paused, total 69143.3s`

We've had a few miscellaneous reports of these types of issues over the years, but it can be difficult to track down what may be causing it. Do you ever see this issue outside of when your machine returns from sleep?

**Comment by arnassavickas** [comment]
Nope, never. It's always when I return to the PC after some time.

**Comment by niik** [comment]
Going to roll this into https://github.com/desktop/desktop/issues/19979 where we've been doing some more troubleshooting into this matter. Please subscribe to that issue to follow along.
